internal/k8s/controllers/node_controller.go

```diff
@@ -65,16 +68,25 @@ func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 }
 
 func (r *NodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	p := predicate.NewPredicateFuncs(
-		func(obj client.Object) bool {
-			node, ok := obj.(*v1.Node)
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
```

Comment:
By doing this, the function will be triggered only during an update because here we are passing the functions explicitly (as opposed to using the predicate.NewPredicateFuncs function).

You need to implement also the other functions (like the add one, the delete one, etc).

Reply:
For the other events I've just restored the previous function which compares nodes' names: I don't know if other checks are needed.